### PR TITLE
Added tests for Transport.java

### PR DESCRIPTION
**Pull request by freddytuxworth**
(Ignore readme changes, branch was synced)

Hi,
I've analysed this project and spotted a few coverage gaps. I've created some tests for Transport.java with the help of Diffblue Cover. Hopefully these tests will help you to spot any regressions caused by future code changes.

Please let me know if you'd like any more tests for other classes.

### CODE REVIEW

File: src/test/java/com/corundumstudio/socketio/TransportTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void byNameIllegalArgument() {
+
+    final String value = "????";
+    thrown.expect(IllegalArgumentException.class);
+    Transport.byName(value);
+
```
Comment (JohnLBergqvist) [inline src/test/java/com/corundumstudio/socketio/TransportTest.java]: Remove the blank lines above & below the method body in each test

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Use the fully-qualified class name in the commit message & PR title, e.g. `com.corundumstudio.socketio.Transport` if the commit message is too long for a single line, trim off the left-most elements until it fits.

File: src/test/java/com/corundumstudio/socketio/TransportTest.java (hunk 1 above)
Comment (petersear) [reply src/test/java/com/corundumstudio/socketio/TransportTest.java]: Agreed

File: src/test/java/com/corundumstudio/socketio/TransportTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import com.corundumstudio.socketio.Transport;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.Assert;
+
+public class TransportTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
```
Comment (petersear) [inline src/test/java/com/corundumstudio/socketio/TransportTest.java]: I'd suggest breaking the line after @Rule as this caused an error for me on a PR and involved another commit/squash/push to resolve. I will log a bug for this.

File: src/test/java/com/corundumstudio/socketio/TransportTest.java (hunk 1 above)
Comment (freddytuxworth) [reply src/test/java/com/corundumstudio/socketio/TransportTest.java]: Done

**Review by paulkennethkent — APPROVED** [review]

**Review by petersear — APPROVED** [review]

**Review by sudo-jaa — APPROVED** [review]
Looks good assuming README changes are rebased or otherwise removed for the full PR :+1:

**Review by JohnLBergqvist — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -48,6 +48,13 @@ Recent Releases
 ================================
 #### Please Note: trunk is current development branch.
 
+#### 11-Jan-2019 - version 1.7.17 released
```
Comment (peterschrammel) [inline README.md:51]: Why is this part of the commit?

**Review by peterschrammel — CHANGES_REQUESTED** [review]
Looks too thin.

File: src/test/java/com/corundumstudio/socketio/TransportTest.java (hunk 2 above)
Comment (petersear) [reply src/test/java/com/corundumstudio/socketio/TransportTest.java]: FYI, I logged this bug:
https://diffblue.atlassian.net/browse/TG-6063

**Pull request closed without merging**